### DISCUSSION TITLE
Programmatic way to replace invalid characters with html

### DISCUSSION THREAD

**Discussion by cysabi** [mdx-js/mdx#2187] [Q&A] [answered]
Hey everyone, I'm looking for a way to automatically replace `<` and `{` etc with html parsed characters programmatically.

The reason is because my users will be able to write markdown, and I don't want to make them put funny symbols in their blog posts because mdx is strict about it.

What's the best way to allow these characters?

![image](https://user-images.githubusercontent.com/47790183/205426213-a7a86a81-f38b-4583-ba13-16436989c65c.png)

**Comment by wooorm** [comment, marked as answer]
You can’t. If you want markdown, configure MDX to parse markdown. If you want MDX, and thus JavaScript, you need to explain to your users that they need to write valid MDX and JS(X).

**Reply by cysabi** [reply]
Hi, can you elaborate when you say "configure MDX to parse markdown"?

From what I was able to gather from the docs, I couldn't find any information on this

**Reply by wooorm** [reply]
- if you want markdown, and have user content, you don’t need MDX at all. Use [unified](https://github.com/unifiedjs/unified) with [remark](https://github.com/remarkjs/remark) or even [micromark](https://github.com/micromark/micromark)
- if you want markdown and MDX, you can use this package, [see the note here for example](https://mdxjs.com/docs/what-is-mdx/), configure the options or configure the file extensions
- if you want MDX, teach your users that they are writing MDX instead of markdown
